### PR TITLE
Check if package-lock.json exists to enable caching

### PR DESCRIPTION
**Pull request by fisehara**
to setup-node with cache

Change-type: patch
Signed-off-by: fisehara <harald@balena.io>

### CODE REVIEW

File: flowzone.yml [hunk 1]
```diff
@@ -1196,17 +1196,33 @@ jobs:
...
-      - name: Setup Node.js
+
+      - name: Check if node-setup can be cached
+        id: cached
+        run: |
+          has_npm_lockfile="$([ -e package-lock.json ] || [ -e npm-shrinkwrap.json ] && echo true || echo false)"
+          echo "has_npm_lockfile=${has_npm_lockfile}" >> $GITHUB_OUTPUT
+    
+      - name: Setup Node.js - Cached
+        if: steps.cached.outputs.has_npm_lockfile  == 'true'
```
Comment (Page-) [inline flowzone.yml:1213]:
```suggestion
        if: steps.cached.outputs.has_npm_lockfile == 'true'
```

File: flowzone.yml [hunk 2]
```diff
@@ -1196,17 +1196,33 @@ jobs:
...
         uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3
         with:
           node-version: "${{ matrix.node_version }}"
           registry-url: "${{ env.NPM_REGISTRY }}"
           cache: "npm"
+      
+      - name: Setup Node.js
+        if: steps.cached.outputs.has_npm_lockfile  != 'true'
```
Comment (Page-) [inline flowzone.yml:1221]:
```suggestion
        if: steps.cached.outputs.has_npm_lockfile != 'true'
```

File: flowzone.yml [hunk 3]
```diff
@@ -1196,17 +1196,33 @@ jobs:
...
       - *downloadSourceArtifact
       - *extractSourceArtifact
 
-      - name: Setup Node.js
+
+      - name: Check if node-setup can be cached
+        id: cached
+        run: |
+          has_npm_lockfile="$([ -e package-lock.json ] || [ -e npm-shrinkwrap.json ] && echo true || echo false)"
+          echo "has_npm_lockfile=${has_npm_lockfile}" >> $GITHUB_OUTPUT
```
Comment (Page-) [inline flowzone.yml:1210]: This should probably be under `npm_check`

Comment (klutchell) [reply flowzone.yml:1210]: +1

Comment (fisehara) [reply flowzone.yml]: moved

**Review by Page- — APPROVED** [review]

**Comment by klutchell** [comment]
LGTM

**Comment by thgreasi** [comment]
:tada:

**Pull request merged**